include/zephyr/modbus/modbus.h

```diff
@@ -384,6 +384,48 @@ int modbus_iface_get_by_name(const char *iface_name);
 typedef int (*modbus_raw_cb_t)(const int iface, const struct modbus_adu *adu,
 				void *user_data);
 
+/**
+ * @brief Custom function code handler function signature.
+ *
+ * Modbus allows user defined function codes which can be used to extend
+ * the base protocol. These callbacks can also be used to implement
+ * function codes currently not supported by Zephyr's Modbus subsystem.
+ *
+ * User data pointer can be used to pass state between subsequent calls to
+ * the handler.
+ *
+ * @param iface      Modbus interface index
+ * @param rx_adu     Pointer to the received ADU struct
+ * @param tx_adu     Pointer to the outgoing ADU struct
+ * @param user_data  Pointer to user data
+ *
+ * @retval           1 If response should be sent, 0 otherwise
+ */
+typedef int (*modbus_user_cb_t)(const int iface,
+				  const struct modbus_adu *rx_adu,
+				  struct modbus_adu *tx_adu,
+				  void *const user_data);
```

Comment:
Nit: Fix indentation

Reply:
Fixed.

Reply:
Please update the commits as per the [contribution guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html). Do not add fixup commits.

I think the PR should contain only 2 commits: 
- Adding the user functions
- Update sample